api_reflector/views.py

```diff
@@ -37,6 +38,19 @@ def match_endpoint(path: str) -> tuple[models.Endpoint, Mapping[str, Any]]:
     # we're disabling mypy here because you're supposed to get strings back from `match`, not full endpoint objects.
     return urls.match(path, method=request.method)  # type: ignore
 
+def _process_error_response(response: Response, ex: UndefinedError | TemplateSyntaxError | TemplateError) -> Response:
+    """Returns the error response for Jinja exceptions"""
+
+    if isinstance(ex, UndefinedError):
+        msg = "Undefined field used in the response template"
+    elif isinstance(ex, TemplateSyntaxError):
+        msg = "Syntax error in response template"
+    else:
+        msg = "Failed to render fields in the response template"
+
+    error_content = f"{msg}. Details: {ex}"
+
+    return Response(error_content, status=500, mimetype=response.content_type)
```

Comment:
```suggestion
    return Response(error_content, status=500, mimetype="text/plain")
```
`response.content_type` is configured in the UI to be able to send back JSON or XML or really any type of data.

for these error messages, we're not sending back that data but instead a plain text error message, so we should set the content type accordingly.

Reply:
good point

Reply:
Updated